Blocks/URL/test.cc

```diff
@@ -265,3 +267,60 @@ TEST(URLTest, EmptyURLException) {
   // Empty host is allowed in relative links.
   EXPECT_EQ("foo://www.website.com:321/second", URL("/second", URL("foo://www.website.com:321/first")).ComposeURL());
 }
+
+namespace url_test {
+
+CURRENT_STRUCT(Simple) {
+  CURRENT_FIELD(a, int64_t);
+  CURRENT_FIELD(b, int64_t);
+  CURRENT_FIELD(s, std::string);
+};
+
+CURRENT_STRUCT(Tricky) {
+  CURRENT_FIELD(s, Optional<std::string>);
+  CURRENT_FIELD(p, (std::pair<std::string, std::string>));
+  CURRENT_FIELD(v, std::vector<std::string>);
+  CURRENT_FIELD(m, (std::map<std::string, std::string>));
+};
+
+}  // namespace url_test
+
+TEST(URLTest, FillsCurrentStructsFromURLParameters) {
+  using namespace url_test;
+
+  {
+    const auto simple = URL("/simple?a=1&b=2&s=test with spaces").query.template FillObject<Simple>();
```

Comment:
The constructor of Current's `URL` class will handle it. (I.e., if you `ComposeURL` this URL back, it'll escape those.)

I suggest we keep it this way.